frontend/src/components/pages/StatisticsPage.tsx

```diff
@@ -0,0 +1,461 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  Box,
+  TextField,
+  Select,
+  MenuItem,
+  InputLabel,
+  FormControl,
+  Typography,
+  Button,
+  Autocomplete,
+  Checkbox,
+  FormControlLabel,
+} from '@mui/material';
+import { BarChart } from '@mui/x-charts/BarChart';
+import { ApiService, ProductRequest } from '@src/api';
+import { useAppDispatch, useAppSelector } from '@src/hooks/ReduxHooks';
+import { categoryToRussian } from '@src/lib/RussianConverters';
+import { Link } from 'react-router-dom';
+import { useQuery, useQueries } from '@tanstack/react-query';
+
+const groupableAttributes: Array<keyof ProductRequest | 'timestamp' | 'user_fullname'> = [
+  'category',
+  'address',
+  'user_fullname',
+  'price',
+  'title',
+  'timestamp',
+];
+
+const attrToRussian: Record<string, string> = {
+  category: 'Категория',
+  address: 'Адрес',
+  user_fullname: 'ФИО пользователя',
+  price: 'Цена',
+  title: 'Название',
+  description: 'Описание',
+  timestamp: 'Дата создания',
+};
+
+const statusOptions = [
+  { value: '', label: 'Все' },
+  { value: 'created_status', label: 'Создана' },
+  { value: 'price_offer_status', label: 'Предложена цена' },
+  { value: 'price_accept_status', label: 'Цена принята' },
+  { value: 'date_offer_status', label: 'Предложена дата' },
+  { value: 'date_accept_status', label: 'Дата принята' },
+  { value: 'closed_status', label: 'Закрыта' },
+];
+
+export default function StatisticsPage() {
+  const [productsData, setProductsData] = useState<ProductRequest[] | null>(null);
+  const [onlyMy, setOnlyMy] = useState(false);
+
+  const fetchData = async (me?: boolean) => {
+    const data = await ApiService.apiRequestsList({ me: me ? true : undefined });
+    setProductsData(data);
```

Comment:
API в твоей ветке так и не перергенерировано. Надо бы это сделать. Заранее могу сказать, что надо будет сделать следующие изменения:

```suggestion
    const { requests } = await ApiService.apiRequestsRetrieve({ me: me ? true : undefined });
    setProductsData(requests);
```